cli/exchange/node.go

```diff
@@ -465,3 +480,341 @@ func NodeListErrors(org string, credToUse string, node string, long bool) {
+	msgPrinter := i18n.GetMessagePrinter()
+	// versionRange is "" for the 1st interation
+	sdef, err := getService(creds, servToCheckDependentService.ServiceOrg, servToCheckDependentService.ServiceURL, servToCheckDependentService.ServiceArch, version, versionRange)
+	if sdef == nil || err != nil {
+		return false, true, errors.New(msgPrinter.Sprintf("service does not exist %v, %v, %v, %v, %v", servToCheckDependentService.ServiceOrg, servToCheckDependentService.ServiceURL, servToCheckDependentService.ServiceArch, version, versionRange))
+	}
+
+	// if the required service has userinputs without default values, then we need to catch that case and issue an error.
+	if !isTopLevelService {
+		if checked, uName := checkUserInputDefaultValueNotEmpty(sdef); !checked {
+			return false, false, errors.New(msgPrinter.Sprintf("required service %v, %v, %v, %v, %v has userinput: %v without default value", servToCheckDependentService.ServiceOrg, servToCheckDependentService.ServiceURL, servToCheckDependentService.ServiceArch, version, versionRange, uName))
+		}
+	}
+
+	dependentServices := sdef.RequiredServices
+	if len(dependentServices) == 0 {
+		return false, true, errors.New(msgPrinter.Sprintf("No dependent service for service %v, %v, %v, %v, %v", servToCheckDependentService.ServiceOrg, servToCheckDependentService.ServiceURL, servToCheckDependentService.ServiceArch, version, versionRange))
+	}
+	for _, dependentService := range dependentServices {
+		if serviceOrg == dependentService.Org && serviceArch == dependentService.Arch && serviceUrl == dependentService.URL {
+			// true if versionRange of the dependent service has intersection with versionRange of service from user input
+			dependentServiceVersionRange := dependentService.VersionRange
+			if &dependentService.VersionRange == nil || dependentService.VersionRange == "" {
+				def := "0.0.0"
+				dependentServiceVersionRange = def
+			}
+
+			// Convert the version to a version expression.
+			dependentServiceVersionExp, err := semanticversion.Version_Expression_Factory(dependentServiceVersionRange)
+			if err == nil {
+				if err = serviceVersionExp.IntersectsWith(dependentServiceVersionExp); err == nil {
+					return true, true, nil
+				}
+			}
+
+		}
+
+		// recursive call accept ServiceReference type
+		depService := ServiceReference{
+			ServiceURL:  dependentService.URL,
+			ServiceOrg:  dependentService.Org,
+			ServiceArch: dependentService.Arch,
+		}
+
+		// recursive call
+		findMatchedService, _, _ := checkDependentServices(creds, serviceOrg, serviceArch, serviceUrl, serviceVersionExp, depService, "", dependentService.VersionRange, false)
+		if findMatchedService {
+			return true, true, nil
+		}
+	}
+
+	return false, true, errors.New(msgPrinter.Sprintf("Service %v, %v, %v, %v, %v does not define as dependent service", serviceOrg, serviceArch, serviceUrl, serviceVersionExp.Get_expression()))
```

Comment:
Please change the wording of this message:
...does not define a dependent service...
Also, I'm not sure this error is ever used. I think it gets ignored. If so, maybe be better to just return nil for it.